### DISCUSSION TITLE
Transmission 4.0 now in Beta

### DISCUSSION THREAD

**Discussion by Skaronator** [haugene/docker-transmission-openvpn#2357] [Ideas]
Just a small heads-up that Transmission 4.0 is now in beta. I think it would be a great way for the community to beta test the new version (especially the rewritten frontend). To make it easier to beta test, it would be great if this Docker Image supported the newly 4.0 beta builds.

https://github.com/transmission/transmission/releases

**Comment by pkishino** [comment]
Yeah, I can try and create a test build in the upcoming days

**Reply by pkishino** [reply]
Sorry, never got around to this as I have hands full currently with all kind of other tasks.
If someone else could try to fork the dev branch and create a 4.0 beta I'd be willing to review and add this to the main repo as a a beta branch possibly.

**Comment by haugene** [comment]
I made an attempt last night, and I have not tested it much but the WebUI loaded at least. I pushed the changes to the `beta` branch.
The [CircleCI build fails](https://app.circleci.com/pipelines/github/haugene/docker-transmission-openvpn?branch=beta) because of a timeout (after 1 hour :scream:). It's not the most beefy nodes we're building on, and we're doing 3 builds in parallel. One per cpu architecture. The "cmake && make && make install" stuff is quite heavy.

Anyways. I built it on my local machine and pushed it so you should be able to try it by running with the beta tag of the image:
```
haugene/transmission-openvpn:beta
```

I think it's great to let the community try out version 4.0.0 through this container image. But:
**DISCLAIMER:** This will not be a widely supported tag and is provided on a best effort basis.

Let me know how it goes if you decide to try it out :smile: :+1: 

EDIT: And it's a huge image. I had to install a bunch of build tools ++. We should do it in a separate build step and many other improvements, but this is a first (and I think working) step.

**Reply by haugene** [reply]
An update. I have been making some progress on the image builds at least. Migrated the build pipeline over to GitHub Actions in #2405 and the timeout limit there is 6 hours and that's enough.

So now the image is at least built for all platforms. Next up is reducing its size.

**Reply by pkishino** [reply]
I’ve been using the beta on two separate containers and it works great.. except magnet links added by remote (sonarr etc) don’t start automatically.. I am tracking the issue in the transmission repo though and created a simple start script which fixes it for now.
Otherwise the beta is great

**Reply by pkishino** [reply]
FYI:
https://github.com/transmission/transmission/issues/3753

**Reply by haugene** [reply]
Cool! I just pushed an updated version of the Dockerfile that runs the cmake && make stuff in a separate build container. Hopefully this reduces the image size a lot :crossed_fingers: 

![image](https://user-images.githubusercontent.com/1038094/201551502-c6225f62-4f64-44db-9c4c-289d9d783389.png)

Right now Docker Hub says we're about 700MB compared to the 100MB of the regular image. The last build on GitHub Actions took 2 hours so let's hope this is also faster :sweat_smile:

**Reply by haugene** [reply]
For future reference the build stuff is stitched together from these sources:
https://github.com/transmission/transmission/discussions/3885
https://github.com/transmission/transmission/issues/4049
https://github.com/transmission/transmission/blob/main/docs/Building-Transmission.md

**Reply by haugene** [reply]
Looking better:
![image](https://user-images.githubusercontent.com/1038094/201619525-f9aae2d4-a059-4722-bf17-5a1a06d7c9e4.png)

FYI @pkishino. I added a section in the main README as well but I'm changing the tag versioning of the images a bit. `latest` is not the latest build on `master` anymore. That's now `edge`. Got away from the custom "calculate image tag" script in CircleCI and using the docker metadata GitHub action instead, it cleans up stuff a bit. `latest` will now be the latest tagged release. Hopefully it will reduce the waves of issues a bit when we merge anything with some bug in it - but mostly we're conforming to some conventions.

**Comment by elisimpson** [comment]
Was able to get this running successfully, including access through RPC via Transmission Remote GUI and a test torrent download. The Transmission version is listed as 3.10.10-dev. Encountered no issues so far.

**Comment by pkishino** [comment]
Just in case anyone needs this, I’m posting my simple start script. Seems in the beta, magnets added remotely(via radarr etc) don’t always start properly.
The script is triggered by cron outside the container, although from inside would work as well. Replace pet and authentication
```
cat transmission_start.sh 
#!/usr/bin/env bash
SESSION_HEADER=$(curl --silent --anyauth --user <user:password> http://localhost:<port>/transmission/rpc/ | sed 's/.*<code>//g;s/<\/code>.*//g')
curl --silent --anyauth --user <user:password> --header "$SESSION_HEADER" http://localhost:<port>/transmission/rpc/ -d "{\"method\":\"torrent-start-now\",\"arguments\": {}}"
```

**Comment by elisimpson** [comment]
Found a bug: the new version rewrites my umask setting from

"umask": 0,

to

"umask": "0",

which then gets misinterpreted by the code as octal "060" instead of "000" which results in weird permissions on some of the folders. Transmission 3 definitely treated the umask setting as a decimal number, not a string. Not sure if this is an intentional change in the new version or a bug.

If it's intentional I should change:

"umask": 0,

to

"umask": "000",

?

**Comment by haugene** [comment]
I have heard that a lot of private trackers block Transmission 4.0, anyone having issues with that?

**Reply by emk2203** [reply]
It's not whitelisted yet on BakaBT, so for me one of the reasons to stay with 3.00 (bb6b5a062e) from your latest image.

**Reply by ilike2burnthing** [reply]
Many private trackers will block anything which isn't a stable release, because it's not worth their time dealing with complaints about inaccurate ratios or H'n'R notices because someone wants to be bleeding edge.

Other PTs use a client whitelist (rather than a blacklist), and betas et al will never get added to it.
